### PR TITLE
refactor(swap): extract swap() transaction build into SwapTransactionBuilder

### LINKED ISSUE #4723: refactor: further reduce SwapFormViewModel.kt (extract swap() transaction build + calculateFees() pipeline)
## Problem

`SwapFormViewModel.kt` is currently ~1,447 lines. #4015 already extracted the quote/cache and gas math into `SwapQuoteManager`, `SwapGasCalculator`, `SwapValidator`, `SwapTokenSelector`, and `SwapDiscountChecker`.

The single largest method that remains is `swap()` (~405 lines): it assembles the swap transaction, runs validation, builds the `SwapTransaction`, and navigates. This is pure transaction-construction logic with no UI-state dependency beyond reading the selected tokens/amount.

**Scope was reduced** (was SCOPE_TOO_LARGE when it also bundled the `calculateFees()` reactive pipeline). This issue is now the low-risk half only. The `calculateFees()` orchestration extraction is tracked separately in #4735.

### Must Do
- [ ] Extract the transaction-building body of `swap()` into a dedicated unit (`SwapTransactionBuilder` / `BuildSwapTransactionUseCase`) that takes the resolved inputs (vault, src/dst tokens, amount, quote, addresses, gas fee) and returns a built `SwapTransaction` or a typed validation error.
- [ ] `swap()` in the ViewModel becomes a thin coordinator: resolve inputs → call the builder → handle result → navigate. Keep the pre-flight validation/resolution and the `addTransaction`/navigate + error handling in the VM (or move validation into the builder behind a typed result).
- [ ] Reuse `SwapGasCalculator`, `approveSpenderFor` (already in `SwapKitDisplay.kt`), and the existing SwapKit `isSignableTxType` guard — do not duplicate them.
- [ ] Add `SwapTransactionBuilderTest` matching the existing pattern (`SwapQuoteManagerTest`, `SwapGasCalculatorTest`) — one case per quote type (ThorChain / MayaChain / SwapKit / OneInch) asserting the built `RegularSwapTransaction` fields/payload.

### Must NOT Do
- Don't change swap behavior, validation messages, or UI.
- Don't modify quote provider integrations.
- Don't touch `calculateFees()` — that's the follow-up issue.

## Acceptance Criteria
- [ ] `swap()` is a thin delegator; heavy tx-build logic lives in `SwapTransactionBuilder`.
- [ ] `SwapTransactionBuilder` has unit-test coverage.
- [ ] `./gradlew assembleDebug` and `./gradlew testDebugUnitTest` succeed.
- [ ] All swap flows (quote, fee display, flip, percentage, execute) work as before.

## Context
**File:** `app/src/main/java/com/vultisig/wallet/ui/models/swap/SwapFormViewModel.kt`
**Target method:** `swap()` (~lines 233–638)
**Prior work:** #4015 (split into `SwapQuoteManager`/`SwapGasCalculator`/etc.); follow-up to #3854, sub-issue of #3852
**Follow-up:** `calculateFees()` reactive quote/fee pipeline extraction → #4735
**Severity:** MEDIUM

## Tasks

- [ ] 1. Extract swap() transaction build into SwapTransactionBuilder and delegate from ViewModel — files: [app/src/main/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilder.kt, app/src/main/java/com/vultisig/wallet/ui/models/swap/SwapFormViewModel.kt]
- [ ] 2. Add SwapTransactionBuilderTest covering ThorChain, MayaChain, SwapKit, and OneInch quote types — files: [app/src/test/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilderTest.kt]

### PR DESCRIPTION
**Pull request by aminsato**
## Summary

Extracts the transaction-construction half of `SwapFormViewModel.swap()` into a dedicated, unit-testable `SwapTransactionBuilder`, as requested in #4723. This is the **low-risk half** of that issue; the `calculateFees()` reactive-pipeline extraction is tracked separately in #4735.

`swap()` carried ~405 lines of provider-specific `when(quote)` logic (ThorChain / MayaChain / SwapKit / 1inch) inside a `viewModelScope.launch`. That body now lives in `SwapTransactionBuilder.build(...)`, which takes the already-resolved inputs and returns a `RegularSwapTransaction`. The ViewModel keeps pre-flight validation/resolution, persistence (`addTransaction`), navigation, and error handling — `swap()` is now a thin delegator.

**`SwapFormViewModel.kt`: 1,447 → 1,185 lines.**

## Behaviour preserved (no functional change)
- SwapKit `isSignableTxType` pre-flight guard
- SwapKit allowance-target (`approveSpenderFor`) derivation
- `estimatedNetworkFeeTokenValue ?: gasFee` and `estimatedNetworkFeeFiatValue ?: gasFeeFiatValue` fee fallbacks (and OneInch's no-fallback `gasFeeFiatValue`)
- 1inch Mantle gas-limit override and Ethereum gas-price patching
- Identical validation messages and navigation

## Tests
- New `SwapTransactionBuilderTest` — one case per quote type (ThorChain / MayaChain / SwapKit / OneInch) asserting dst-address derivation, approval logic, fee fallbacks, and payload type, plus the SwapKit unsupported-`txType` guard.
- `SwapFormViewModelTest` updated for the new constructor (wires a real builder).
- `./gradlew :app:testDebugUnitTest --tests "com.vultisig.wallet.ui.models.swap.*"` ✅
- `./gradlew assembleDebug` ✅

## Notes
This was originally assigned to the repo agent on #4723, but its planning step stalled repeatedly on a decompose loop, so the change was implemented manually. No UI is touched, so no before/after screenshots apply.

Closes #4723

## Summary by CodeRabbit

* **Refactor**
  * Internal swap transaction assembly delegated to a new, injected builder to simplify internals; end-to-end swap behavior unchanged for users.
* **Tests**
  * Expanded unit coverage for swap transaction assembly across providers and edge cases, including fee handling and routing scenarios.

### CODE REVIEW

File: app/src/test/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilderTest.kt [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        coEvery { allowanceRepository.getAllowance(any(), any(), any(), any()) } returns null
+    }
+
+    @Test
+    fun `builds ThorChain swap with inbound dst address and estimated-fee fallbacks`() = runTest {
+        val srcToken = coin(Chain.Bitcoin, "BTC", "bc1qsrc", 8, isNative = true)
+        val dstToken = coin(Chain.Ethereum, "ETH", "0xdst", 18, isNative = true)
+        val data =
+            mockk<THORChainSwapQuote>(relaxed = true).apply {
+                every { router } returns null
```
Comment (rkokhatskyi) [inline app/src/test/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilderTest.kt:69]: Every branch test uses a native source with `router` null, so the `router` priority for `dstAddress`, the `isRouterDeposit` true path that forwards `dstAddress`, `memo` and `tokenAmountValue` into `getSpecificAndUtxo`, and MayaChain's EVM source `dstAddress` branch all go unexercised, while the absence of any `coVerify` on the `getSpecificAndUtxo` arguments means the same `specificAndUtxo` is asserted no matter the inputs so inverting `isRouterDeposit` would still pass, whereas an EVM source case with `router` set plus a `coVerify` on the forwarded arguments would cover the routing

File: app/src/test/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilderTest.kt [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+                estimatedNetworkFeeTokenValue = null,
+                estimatedNetworkFeeFiatValue = null,
+            )
+
+        assertEquals("maya-inbound", tx.dstAddress)
+        // No estimated fee available -> falls back to the passed gasFee / gasFeeFiatValue
+        assertEquals(TokenValue(BigInteger.valueOf(4), srcToken), tx.gasFees)
+        assertEquals(FiatValue(BigDecimal("0.10"), "USD"), tx.gasFeeFiatValue)
+        assertFalse(tx.isApprovalRequired)
+        assertIs<SwapPayload.MayaChain>(tx.payload)
```
Comment (rkokhatskyi) [inline app/src/test/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilderTest.kt]: The payload checks stop at `assertIs<...>(tx.payload)` so the distinguishing fields are never asserted, which leaves `streamingInterval` (`"1"` for ThorChain versus `"3"` for MayaChain), `isAffiliate`, `toAmountLimit` and `expirationTime` unverified so a copy paste that swapped the streaming interval between the two branches would still pass, so casting the payload and asserting those fields would lock them in

File: app/src/test/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilderTest.kt [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+            assertEquals("0xrouter", tx.dstAddress)
+            // The ERC20 approve spender is the dedicated allowance target, not the swap `to`.
+            assertEquals("0xproxy", tx.approveSpender)
+            assertTrue(tx.isApprovalRequired)
+            // OneInch always uses the passed gasFeeFiatValue (no estimated-fee fallback).
+            assertEquals(FiatValue(BigDecimal("2.00"), "USD"), tx.gasFeeFiatValue)
+            // gasFees still falls back to gasFee when no estimated network fee is available.
+            assertEquals(TokenValue(BigInteger.valueOf(8), srcToken), tx.gasFees)
+            assertIs<SwapPayload.EVM>(tx.payload)
```
Comment (rkokhatskyi) [inline app/src/test/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilderTest.kt]: The shared `specificAndUtxo` fixture stubs `blockChainSpecific` with a relaxed `mockk` of the sealed parent `BlockChainSpecific`, which is never a `BlockChainSpecific.Ethereum`, so the `if (specific is BlockChainSpecific.Ethereum)` branch in `build()` is skipped, and with no `Chain.Mantle` case and no assertion on the resulting `SwapPayload.EVM` `quote.tx` the 1inch gas price patch and the Mantle gas limit override carry zero coverage despite the description listing them as preserved, whereas a fixture using a real `BlockChainSpecific.Ethereum` plus a `Chain.Mantle` case asserting `tx.gasPrice` and `tx.gas` would exercise both

File: app/src/main/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilder.kt [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+            is SwapQuote.SwapKit -> {
+                // Pre-flight gate: refuse a route whose txType has no wired signing
+                // path before staging keysign. Sourced from
+                // SwapKitSwapPayloadJson.SIGNABLE_TX_TYPES — the same list
+                // SigningHelper dispatches on — so this guard can't drift from what
+                // the dispatcher actually accepts.
+                require(SwapKitSwapPayloadJson.isSignableTxType(quote.data.txType)) {
+                    "Unsupported SwapKit txType for swap: ${quote.data.txType}"
+                }
+                val specificAndUtxo =
```
Comment (rkokhatskyi) [inline app/src/main/java/com/vultisig/wallet/ui/models/swap/SwapTransactionBuilder.kt:196]: This branch resolves `blockChainSpecific` through `getSpecificAndUtxo`, yet the comment just below states signing for SwapKit is driven entirely by the payload bytes, so the call only populates a field the SwapKit signer never reads while adding an RPC failure surface where a flaky nonce or UTXO lookup can fail an otherwise valid swap, which is carried over verbatim so it sits outside this refactor but is worth a follow up to skip the lookup for SwapKit

**Review by rkokhatskyi — CHANGES_REQUESTED** [review]

**Comment by aminsato** [comment]
@Vaulty-bot pick-up

**Review by rkokhatskyi — APPROVED** [review]
LGTM
